dkg/nodesigs.go

```diff
@@ -84,22 +158,25 @@ func (n *nodeSigBcast) exchange(
 		return nil, errors.Wrap(err, "k1 lock hash signature broadcast")
 	}
 
-	ret := make([][]byte, n.operatorAmt)
-	ret[nodeIdx.PeerIdx] = sig
+	n.otherSigs[n.nodeIdx.PeerIdx] = localSig
+
+	tick := time.NewTicker(100 * time.Millisecond)
+
+	for {
+		var done bool
 
-	// see newLockHashK1Bcast comment
-	for idx := 0; idx < n.operatorAmt-1; idx++ {
 		select {
 		case <-ctx.Done():
-			return nil, errors.New("context canceled")
-		case otherSig := <-n.otherSigs:
-
-			// TODO: remove if appears too verbose down the line
-			log.Debug(ctx, "Received node signature", z.Uint("sender", uint(otherSig.GetPeerIndex())))
+			return nil, ctx.Err()
+		case <-tick.C:
+			done = n.allSigs()
```

Comment:
can inline:
```
if n.allsigs() {
  tick.Stop()
  break
}
```

Reply:
written like this the break will only get out of the `select`, not the `for` as well - i could use `goto` but i'd rather leave it like this

Reply:
ok cool